sdk/search/azure-search-documents/src/test/java/com/azure/search/documents/FieldBuilderTest.java

```diff
@@ -0,0 +1,98 @@
+package com.azure.search.documents;
+
+import com.azure.search.documents.models.AnalyzerName;
+import com.azure.search.documents.models.ComplexField;
+import com.azure.search.documents.models.DataType;
+import com.azure.search.documents.models.Field;
+import com.azure.search.documents.models.SearchableField;
+import com.azure.search.documents.models.SimpleField;
+import com.azure.search.documents.test.environment.models.Hotel;
+import com.azure.search.documents.test.environment.models.HotelSearchException;
+import com.azure.search.documents.test.environment.models.HotelSearchableExceptionOnList;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class FieldBuilderTest {
+    @Test
+    public void hotelComparison() {
+        List<Field> actualFields = sortByFieldName(FieldBuilder.build(Hotel.class));
+        List<Field> expectedFields = sortByFieldName(buildHotelFieldsFromModel());
+        assertEquals(expectedFields.size(), actualFields.size());
+        for (int i = 0; i < expectedFields.size(); i++) {
+            TestHelpers.assertObjectEquals(expectedFields.get(i), actualFields.get(i));
+        }
+    }
+
+    @Test
+    public void hotelSearchableThrowException() {
+        Exception exception = assertThrows(RuntimeException.class, () -> {
+            FieldBuilder.build(HotelSearchException.class);
+        });
+        assertTrue(exception.getMessage().contains("hotelId"));
+        assertTrue(exception.getMessage().contains(DataType.EDM_INT32.toString()));
+    }
+
+    @Test
+    public void hotelListFieldSearchableThrowException() {
+        Exception exception = assertThrows(RuntimeException.class, () -> {
+            FieldBuilder.build(HotelSearchableExceptionOnList.class);
+        });
+        assertTrue(exception.getMessage().contains("passcode"));
+        assertTrue(exception.getMessage().contains(DataType.collection(DataType.EDM_INT32).toString()));
+    }
+
+    private List<Field> buildHotelFieldsFromModel() {
```

Comment:
Nit: wouldn't "build...FromModel" mean using the field builder against a model type? This seems more like manual index creation / defining of fields.

Reply:
Will rename